modules/sponsorship.py

```diff
@@ -702,6 +702,18 @@ def sponsor_verify_url(request):
 
     return URL.verify(request, hmac_key=sponsor_hmac_key())
 
+def sponsor_signed_url(username, *args):
+    """
+    Return a URL. args passed are sent to the URL method (e.g. page; or controller, page)
+    """
+    URL = current.globalenv['URL']
+    return URL(
+        *args,
+        args=[username],
+        scheme=True,
+        host=True,
+        hmac_key=sponsor_hmac_key()
+    )
```

Comment:
Having this as a private function was an intentional choice. Despite initial appearances, there isn't much about this function that is globally applicable:

* It assumes that the URL will have ``username`` appended to it, so only relevant to the 2 pages we currently hand to it.
* It assumes we need a fully-specified URL, as we do for e-mails. You wouldn't set scheme/host for use in a web-page (although I'm not sure why you'd want to do that).

If you removed these restrictions, then there's basically nothing left in this function. You might as well call ``URL``() directly.

Also, being private removes any mental load of wondering whether ``sponsor_signed_url``() is used elsewhere if you choose to rewrite it.

This is all incredibly minor stuff though.

Reply:
Ah, I was about to use it in a management page, so that I could have a page to look at that gives the URLs that a user will be emailed. Does that make sense?

Reply:
Ah, fair do's. I think my preference would be to go all-out and call ``sponsorship_email_reminders([username_on_my_page])`` to get the URLs that would go on this page. Obviously excessive, but:

* It's a management page, so excess is fine :)
* It might be interesting to know what OTTs they have expiring on such a page anyway, so the DB lookup doesn't have to be wasted.
* There's less code that needs to stay in lock-step, e.g. if we change the endpoint that does renewal, we'd have to modify both ``sponsorship_email_reminders`` and the management page with the new file-name.

OTOH, there's probably a reason you've not done that. In which case you might as well expose them as the commit does.

Reply:
Ah, you're right, `sponsorship_email_reminders` is probably better. I'll revert.

Reply:
Thanks for responding to this BTW

Reply:
One issue is that if this is called from the management page, the URL is wrong, because it uses the "manage" controller. So we should probably hard-code the controller to be "default" in `sponsor_signed_url`

Reply:
No worries!

Reply:
> So we should probably hard-code the controller to be "default"

Yup, that's definitely something that needs doing. I thought I already did, but must be thinking of something else.

Reply:
Done and reworded